falcon/util/structures.py

```diff
@@ -27,17 +27,17 @@
 
 """
 
-import collections
+from collections.abc import Mapping, MutableMapping
```

Comment:
Hi @aparkerlue !
And thanks for your effort to rectify this. However, it seems we will need a more sophisticated approach here since we are still supporting Python 2.7 and PyPy2.

On Python 2:
```python
>>> from collections.abc import Mapping, MutableMapping
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named abc
```

See also the CI reports.

Could we consider a some sort of conditional import based on Python version, or ``try`` ... ``except ImportError`` approach?

Reply:
I went ahead and addressed this by refactoring the change to use the new ``compat`` module. Let me know what you think.